core/sys_context.go

```diff
@@ -0,0 +1,83 @@
+package core
+
+import (
+	"math/big"
+
+	"github.com/celo-org/celo-blockchain/common"
+	"github.com/celo-org/celo-blockchain/contracts/blockchain_parameters"
+	"github.com/celo-org/celo-blockchain/contracts/currency"
+	"github.com/celo-org/celo-blockchain/contracts/gasprice_minimum"
+	"github.com/celo-org/celo-blockchain/core/vm"
+)
+
+// SysContractCallCtx represents a system contract call context for a given block (represented by vm.EVMRunner).
+// It MUST sit on the header.Root of a block, which is parent of the block we intend to deal with.
+type SysContractCallCtx struct {
+	vmRunner vm.EVMRunner // vmRunner represents the parent block state on which the contract calls will be made
+
+	whitelistedCurrencies     map[common.Address]struct{}
+	gasForAlternativeCurrency uint64
+	// gasPriceMinimums stores values for whitelisted currencies keyed by their contract address
+	// Note that native token(CELO) is keyed by common.ZeroAddress
+	gasPriceMinimums map[common.Address]*big.Int
+}
+
+// NewSysContractCallCtx creates the SysContractCallCtx object and makes the contract calls.
+func NewSysContractCallCtx(vmRunner vm.EVMRunner) (sc *SysContractCallCtx) {
+	sc = &SysContractCallCtx{
+		vmRunner:              vmRunner,
```

Comment:
The VM runner isn't used after the construction of the `SysContractCallCtx` so it doesn't need to be kept around.